wrappers/manpages.go

```diff
@@ -0,0 +1,104 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package wrappers
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/snap"
+	"syscall"
+)
+
+func AddSnapManpages(s *snap.Info) error {
+	snapManRoot := filepath.Join(s.MountDir(), "meta", "man")
+	inSnapPrefix := s.SnapName() + "."
+	onDiskPrefix := s.InstanceName() + "."
+	return filepath.Walk(snapManRoot, func(path string, info os.FileInfo, err error) error {
```

Comment:
As I understand the first `if` is suppose to handle the case when there's no man directory inside snap meta, but as a side effect it swallows up all errors. I think it'd be better to have:

```
if !osutil.DirExists(snapManRoot) {
	return nil
}
return filepath.Walk(snapManRoot, func(path string, info os.FileInfo, err error) error {
	if err != nil {
		return err
	}
	...
}
```